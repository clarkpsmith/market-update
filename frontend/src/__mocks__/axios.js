const BASE_URL = "https://apidojo-yahoo-finance-v1.p.rapidapi.com";
import {
  generalNewsData,
  tslaChartData,
  tslaData,
  tslaNews,
  indexData,
  spData,
  spChartData,
} from "../testValues";

export default {
  get: jest.fn().mockImplementation((url) => {
    switch (url) {
      case `${BASE_URL}/market/v2/get-quotes?region=us&symbols=tsla`:
        return Promise.resolve(tslaData);
      case `${BASE_URL}/market/v2/get-quotes?region=us&symbols=TSLA`:
        return Promise.resolve(tslaData);
      case `${BASE_URL}/market/v2/get-quotes?region=us&symbols=^gspc`:
        return Promise.resolve(spData);
      case "https://apidojo-yahoo-finance-v1.p.rapidapi.com/market/v2/get-quotes?region=us&symbols=^gspc,^dji,^ixic,^rut,^tnx,usdeur=x,gc=f,cl=f,btc-usd":
        return Promise.resolve(indexData);
      case `${BASE_URL}/market/get-charts?region=us&symbol=^GSPC&interval=1d&range=2y`:
        return Promise.resolve(spChartData);
      case `${BASE_URL}/market/get-charts?region=us&symbol=tsla&interval=1d&range=2y`:
        return Promise.resolve(tslaChartData);
      case `${BASE_URL}/market/get-charts?region=us&symbol=TSLA&interval=1d&range=2y`:
        return Promise.resolve(tslaChartData);
      case "https://yahoo-finance15.p.rapidapi.com/api/yahoo/ne/news":
        return Promise.resolve(generalNewsData);
      case "https://yahoo-finance15.p.rapidapi.com/api/yahoo/ne/news/tsla":
        return Promise.resolve(tslaNews);
      case "https://yahoo-finance15.p.rapidapi.com/api/yahoo/ne/news/TSLA":
        return Promise.resolve(tslaNews);
      case "http://localhost:3001/users/testuser":
        return Promise.resolve({
          data: {
            user: {
              username: "testuser",
              firstName: "test",
              lastName: "user",
              email: "testuser@gmail.com",
              isAdmin: false,
              watchlist: ["TSLA"],
            },
          },
        });
      case "http://localhost:3001/users/testuser2":
        return Promise.resolve({
          data: {
            user: {
              username: "testuser2",
              firstName: "firstname",
              lastName: "lastname",
              email: "testuser2@gmail.com",
              isAdmin: false,
              watchlist: [],
            },
          },
        });
      default:
        throw new Error(`UNMATCHED URL: ${url}`);
    }
  }),

  post: jest.fn().mockImplementation((url) => {
    switch (url) {
      case "http://localhost:3001/auth/token":
        return Promise.resolve({
          data: {
            token:
              "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyIiwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTYyMTAyMDU2NH0.BIb3UZznOFTq_MwmahBVAQuTmFAuJVRnp8sWEUyi3I4",
          },
        });
      case "http://localhost:3001/auth/register":
        return Promise.resolve({
          data: {
            token:
              "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyMiIsImlzQWRtaW4iOmZhbHNlLCJpYXQiOjE2MjE4NzUzNDh9.VkwYtdA-NQ1OkiJBrhaWEcOnvwvBmfduNXufkgWEXLY",
          },
        });
      default:
        throw new Error(`UNMATCHED URL: ${url}`);
    }
  }),
  patch: jest.fn().mockImplementation((url) => {
    switch (url) {
      case "http://localhost:3001/users/testuser":
        return Promise.resolve({
          data: {
            user: {
              firstName: "firstchanged",
              lastName: "user",
              email: "testuser@gmail.com",
              username: "testuser",
              watchlist: ["tsla"],
              isAdmin: false,
            },
          },
        });

      default:
        throw new Error(`UNMATCHED URL: ${url}`);
    }
  }),
};
