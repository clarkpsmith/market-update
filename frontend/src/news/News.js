import React, { useEffect, useState } from "react";
import YahooFinanceApi from "../api/YahooFinanceApi";
import Article from "./Article";
import ChaseLoading from "../chaseloading/ChaseLoading";

const News = ({ ticker }) => {
  const [stockNews, setStockNews] = useState(null);
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    async function getNews(ticker) {
      const res = await YahooFinanceApi.getStockNews(ticker);
      setStockNews(() => res);
      setLoading(false);
    }

    getNews(ticker);
  }, []);

  if (loading) {
    return (
      <div className="Watchlist">
        <ChaseLoading />
      </div>
    );
  }

  const articles = [];

  for (let i = 0; i < 5; i++) {
    articles.push(<Article key={stockNews[i].link} data={stockNews[i]} />);
  }

  return <div className="News">{articles}</div>;
};

export default News;
