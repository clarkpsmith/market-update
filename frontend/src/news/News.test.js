import React from "react";
import axios from "axios";
import { render, waitForElementToBeRemoved } from "@testing-library/react";
import News from "./News";
import { tslaData } from "../testValues";

it("renders without crashing", () => {
  render(<News ticker="tsla" />);
});

it("matches snapshot", async () => {
  const { asFragment, queryByTestId } = render(<News ticker="tsla" />);
  await waitForElementToBeRemoved(() => queryByTestId("loading"));

  expect(asFragment()).toMatchSnapshot();
});
