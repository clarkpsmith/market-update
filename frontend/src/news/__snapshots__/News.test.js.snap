// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="News"
  >
    <div
      class="Article"
    >
      <a
        aria-label="See Article"
        class="Article-link"
        href="https://finance.yahoo.com/news/tesla-inc-tsla-still-great-170544919.html?.tsrc=rss"
      >
        <div
          class="Article-date"
        >
           Thu, 13 May 2021
        </div>
      </a>
      <h5
        class="Article-title"
      >
        Is Tesla Inc. (TSLA) Still A Great Investment Choice?
      </h5>
      <div
        class="Article-description"
      />
    </div>
    <div
      class="Article"
    >
      <a
        aria-label="See Article"
        class="Article-link"
        href="https://finance.yahoo.com/news/tesla-staring-deepest-weekly-drop-164056074.html?.tsrc=rss"
      >
        <div
          class="Article-date"
        >
           Thu, 13 May 2021
        </div>
      </a>
      <h5
        class="Article-title"
      >
        Tesla Staring at Deepest Weekly Drop Since Virus-Fueled Selloff
      </h5>
      <div
        class="Article-description"
      />
    </div>
    <div
      class="Article"
    >
      <a
        aria-label="See Article"
        class="Article-link"
        href="https://finance.yahoo.com/news/1-musk-decries-bitcoins-insane-162319689.html?.tsrc=rss"
      >
        <div
          class="Article-date"
        >
           Thu, 13 May 2021
        </div>
      </a>
      <h5
        class="Article-title"
      >
        UPDATE 1-Musk decries bitcoin's 'insane' energy use after Tesla payment U-turn
      </h5>
      <div
        class="Article-description"
      />
    </div>
    <div
      class="Article"
    >
      <a
        aria-label="See Article"
        class="Article-link"
        href="https://finance.yahoo.com/m/10b712f4-13ec-3079-a9fb-dcc85ae15a2c/tesla-stock-breaks-support%2C.html?.tsrc=rss"
      >
        <div
          class="Article-date"
        >
           Thu, 13 May 2021
        </div>
      </a>
      <h5
        class="Article-title"
      >
        Tesla Stock Breaks Support, Bitcoin Price Tumbles After Elon Musk Halts Cryptocurrency Use For Vehicle Purchases
      </h5>
      <div
        class="Article-description"
      />
    </div>
    <div
      class="Article"
    >
      <a
        aria-label="See Article"
        class="Article-link"
        href="https://finance.yahoo.com/news/grayscale-ceo-sees-etf-solution-161812323.html?.tsrc=rss"
      >
        <div
          class="Article-date"
        >
           Thu, 13 May 2021
        </div>
      </a>
      <h5
        class="Article-title"
      >
        Grayscale CEO Sees ETF as Solution to Biggest Crypto Fundâ€™s Woes
      </h5>
      <div
        class="Article-description"
      />
    </div>
  </div>
</DocumentFragment>
`;
